src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,473 @@
+            undefined;
+        Resp ->
+            lager:error("Unknown response from RabbitMQ management console: ~p", [Resp]),
+            undefined
+
+    end.
+
+% make an http connection to the rabbitmq management console
+% and return a integer value or undefined
+-spec get_current_length() -> integer() | undefined.
+get_current_length() ->
+    CurrentResult = rabbit_mgmt_server_request(?QUEUE_LENGTH_PATH),
+    case CurrentResult of
+        {error, {conn_failed,_}} ->
+            lager:info("Can't connect to RabbitMQ management console"),
+            undefined;
+        {ok, "200", _, CurrentStatusJson} ->
+            parse_current_length_response(CurrentStatusJson);
+        {ok, "404", _, _} ->
+            lager:info("Queue not bound in /analytics exchange"),
+            undefined;
+        Resp ->
+            lager:error("Unknown response from RabbitMQ management console: ~p", [Resp]),
+            undefined
+    end.
+
+% NOTE: oc_httpc:responseBody() :: string() | {file, filename()}.
+% reach into the JSON returned from the RabbitMQ management console
+% and return a current length value, OR undefined if unavailable or
+% unparseable. EJ was not convenient for parsing this data.
+-spec parse_current_length_response(binary() | {file, oc_httpc:filename()}) -> integer() | undefined.
+parse_current_length_response(Message) ->
+    try
+        CurrentJSON = jiffy:decode(Message),
+        % make a proplists of each queue and it's current length
+        QueueLengths =
+            lists:map(fun (QueueStats) -> {QS} = QueueStats,
+                                        {proplists:get_value(<<"name">>, QS),
+                                        proplists:get_value(<<"messages">>, QS)}
+                    end, CurrentJSON),
+        % look for the alaska queue length
+        parse_integer(proplists:get_value(<<"alaska">>, QueueLengths, undefined))
+    catch
+        Error:Rsn ->
+            lager:error("Invalid RabbitMQ response while getting queue length ~p ~p",
+                                 [Error, Rsn]),
+            undefined
+    end.
+
+% NOTE: oc_httpc:responseBody() :: string() | {file, filename()}.
+% reach into the JSON returned from the RabbitMQ management console
+% and return the max_length value, OR undefined if unavailable or
+% unparseable. EJ was not convenient for parsing this data.
+-spec parse_max_length_response(binary() | {file, oc_httpc:filename()}) -> integer() | undefined.
+parse_max_length_response(Message) ->
+    try
+        {MaxLengthPolicy} = jiffy:decode(Message),
+        {Defs} = proplists:get_value(<<"definition">>, MaxLengthPolicy),
+        parse_integer(proplists:get_value(<<"max-length">>, Defs, undefined))
+    catch
+        Error:Rsn->
+            lager:error("Invalid RabbitMQ response while getting queue max length ~p ~p",
+                        [Error, Rsn]),
+            undefined
+    end.
+
+
+-spec parse_integer(any()) -> integer | undefined.
```

Comment:
I think we have one or two other safe-list-to-int-converters kicking around here somewhere.